packages/common/src/state/simple.ts

```diff
@@ -0,0 +1,185 @@
+import { Account, bytesToHex } from '@ethereumjs/util'
+import { keccak256 } from 'ethereum-cryptography/keccak.js'
+
+import { OriginalStorageCache } from './originalStorageCache.js'
+
+import type {
+  AccountFields,
+  EVMStateManagerInterface,
+  Proof,
+  StorageDump,
+  StorageRange,
+} from '../interfaces.js'
+import type { Address, PrefixedHexString } from '@ethereumjs/util'
+
+/**
+ * Simple and dependency-free state manager for basic state access use cases
+ * where a trie or verkle backed state manager is too heavy-weight.
+ *
+ * This state manager comes with the basic state access logic for
+ * accounts, storage and code (put* and get* methods) as well as a simple
+ * implementation of checkpointing but lacks methods implementations of
+ * state root related logic as well as some other non-core functions.
+ *
+ * Functionality provided is sufficient to be used for simple EVM use
+ * cases and the state manager is used as default there.
+ *
+ * For a more full fledged and MPT-backed state manager implementation
+ * have a look at the `@ethereumjs/statemanager` package.
+ */
+export class SimpleStateManager implements EVMStateManagerInterface {
+  public accountStack: Map<PrefixedHexString, Account | undefined>[] = []
+  public codeStack: Map<PrefixedHexString, Uint8Array>[] = []
+  public storageStack: Map<string, Uint8Array>[] = []
+
+  originalStorageCache: {
+    get(address: Address, key: Uint8Array): Promise<Uint8Array>
+    clear(): void
+  }
+
+  constructor() {
+    this.add()
+    this.originalStorageCache = new OriginalStorageCache(this.getContractStorage.bind(this))
+  }
+
+  topA() {
+    return this.accountStack[this.accountStack.length - 1]
+  }
+  topC() {
+    return this.codeStack[this.codeStack.length - 1]
+  }
+  topS() {
+    return this.storageStack[this.storageStack.length - 1]
+  }
+
+  add() {
+    const newTopA = new Map(this.topA())
+    for (const [address, account] of newTopA) {
+      const accountCopy =
+        account !== undefined
+          ? Object.assign(Object.create(Object.getPrototypeOf(account)), account)
+          : undefined
+      newTopA.set(address, accountCopy)
+    }
+    this.accountStack.push(newTopA)
+    this.codeStack.push(new Map(this.topC()))
+    this.storageStack.push(new Map(this.topS()))
+  }
```

Comment:
What does this method do? It seems to copy the account, but it shallow copies the code and storage stack?

Reply:
(is this a checkpoint?)

Reply:
Ah, I think I can directly call `this.commit()` from within the constructor and put this code there.

Will try, then we can skip the extra method.

Reply:
Ah, now I remember, I added this since I otherwise couldn't call the async `checkpoint()` method from the constructor.

I changed to a more expressive name (`checkpointSync()`), if there is a more elegant solution let me know.